connman/vpn/plugins/wireguard.c

```diff
@@ -346,6 +446,13 @@ static void run_dns_reresolve(struct wireguard_info *info)
 	if (info->reresolve_id)
 		g_source_remove(info->reresolve_id);
 
+	if (vpn_provider_get_connection_errors(info->provider) >=
+						DNS_RERESOLVE_ERROR_LIMIT) {
+		connman_warn("reresolve error limit reached");
+		vpn_died(NULL, -ENONET, info->provider);
+		return;
```

Comment:
if reresolve_id was nonzero and thus removed, this early return leaves stale source id behind

Reply:
Yeah, made this better with 9ce8543874c54a9b936850b0bdb1d9c1bf6d6cdb